src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/JndiDirectoryDAO.scala

```diff
@@ -135,55 +128,61 @@ class JndiDirectoryDAO(protected val directoryConfig: DirectoryConfig)(implicit
           Seq("top", "inetOrgPerson").foreach(oc.add)
           myAttrs.put(oc)
 
-          myAttrs.put(new BasicAttribute(Attr.email, email.value))
-          myAttrs.put(new BasicAttribute(Attr.sn, subject.value))
-          myAttrs.put(new BasicAttribute(Attr.cn, subject.value))
-          myAttrs.put(new BasicAttribute(Attr.uid, subject.value))
+          myAttrs.put(new BasicAttribute(Attr.email, user.email.value))
+          myAttrs.put(new BasicAttribute(Attr.sn, user.id.value))
+          myAttrs.put(new BasicAttribute(Attr.cn, user.id.value))
+          myAttrs.put(new BasicAttribute(Attr.uid, user.id.value))
 
           myAttrs
         }
       }
 
-      ctx.bind(subjectDn(subject), identityContext)
+      ctx.bind(subjectDn(user.id), identityContext)
+      user
     } catch {
       case _: NameAlreadyBoundException =>
-        throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"identity with id ${subject.value} already exists"))
+        throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"identity with id ${user.id} already exists"))
     }
   }
 
-  override def getPetServiceAccountForUser(userId: WorkbenchUserId): Future[Option[WorkbenchUserServiceAccountEmail]] = {
```

Comment:
pet SAs implicitly belong to a user now. Loading a pet SA implies a user so no need for this function. Also no need for special add/remove below